### PR TITLE
Marginal Sprite Improvements Part 2: Cigarette Lighter Boogaloo

### PR DESCRIPTION
**Pull request by JStheguy**
[Changelogs]:
:cl: Big Tobacco
add: Cheap lighters now come in four different shapes and Zippos can have one of four different designs, collect them all and improve our sales!
imageadd: All existing lighter sprites have been given a face-lift and the flames are now subtly animated.
tweak: Cheap lighters now use a fixed list of 16 colors to pick from instead of using totally randomized colors.
imageadd: Cigar cases have been tweaked to be a bit smaller and to have a modified design.
imageadd: Cohiba Robusto and Havanan cigars now use separate case sprites from the generic premium cigar cases.
/:cl:

[why]: The current slime Zippo is ugly as hell and the current method of color selection for cheap lighters can produce questionable results. The rest is just three tons of icing on the cupcake.

New Sprites
image

Old Sprites
image

Flame Animation
image

### CODE REVIEW

**Comment by Mickyan** [comment]
My only pet peeve is the inside of the cigar case blends with the brown of the cigars a bit too much, you can see how in the old sprites they pop out way more clearly

I'd say make the grey one darker and the blue one lighter

**Comment by DaxDupont** [comment]
why are the lighters so round

**Comment by Ratvar** [comment]
Grey zippo changes to green after unflip?

**Comment by Wilchenx** [comment]
@DaxDupont my lighter is round and full

**Comment by epochayur** [comment]
i'm gonna miss that old zippo sprite

**Review by Jalleo — CHANGES_REQUESTED** [review]
Sorry for the bikeshedding but I find the blue well uh off. Could you try a red or a purple colour instead?

**Comment by JStheguy** [comment]
@Ratvar 
They're different lighters with different colors.

**Comment by JStheguy** [comment]
image
Also, tweaked the interiors of the cigar cases, better?

**Comment by KomradeSpectre** [comment]
Whereas I agree with the changes on the cheap lighters, it seems like the Zippos lost a good chunk of detail. Particularly on the chimney of it where it was dithered previously. Also the flames are too sharp now, they were soft before, but the colors are better in your version.

**Comment by ShizCalev** [comment]
#31871 +1

File: code/game/objects/items/storage/fancy.dm [hunk 1]
```diff
@@ -293,22 +293,24 @@
 /obj/item/storage/fancy/cigarettes/cigars/update_icon()
 	cut_overlays()
 	if(fancy_open)
-		add_overlay("[icon_state]_open")
+		icon_state = "[initial(icon_state)]_open"
 		var/mutable_appearance/cigar_overlay = mutable_appearance(icon, icon_type)
 		for(var/c = contents.len, c >= 1, c--)
-			cigar_overlay.pixel_x = 4 * (c - 1)
+			cigar_overlay.pixel_x = 3 * (c - 1)
```
Comment (ShizCalev) [inline code/game/objects/items/storage/fancy.dm]: @JStheguy if you want to take a pop at #15492, all you have to do is make a sprite with all the cigars in the case, then replace this all with 

```
if(contents.len)
	var/mutable_appearance/cigar_overlay = mutable_appearance(icon, "icon_type_[contents.len]")
	add_overlay(cigar_overlay)
```

**Pull request merged**